zerver/lib/markdown/__init__.py

```diff
@@ -1812,13 +1818,16 @@ def handleMatch(  # type: ignore[override] # https://github.com/python/mypy/issu
                 return None, None, None
 
             el = Element("span")
-            el.set("data-user-id", user_id)
+            if user_id:
+                el.set("data-user-id", user_id)
             text = f"{name}"
             if silent:
                 el.set("class", "user-mention silent")
             else:
                 el.set("class", "user-mention")
                 text = f"@{text}"
+            if topic_wildcard:
+                el.set("class", el.get("class", "") + " topic-mention")
```

Comment:
I'm not sure we want `user-mention` as well in this case; we can discuss here:

https://chat.zulip.org/#narrow/stream/378-api-design/topic/.40topic-mention.20data-user-id.20.2326195/near/1608185

Reply:
Removed `user-mention` after discussion in CZO